model/common/src/icon4py/model/common/grid/vertical.py

```diff
@@ -517,3 +532,133 @@ def get_vct_a_and_vct_b(vertical_config: VerticalGridConfig) -> tuple[fa.KField,
         if vertical_config.file_path
         else _compute_vct_a_and_vct_b(vertical_config)
     )
+
+
+def init_vert_coord(
+    vct_a: fa.KField[ta.wpfloat],
+    topography: fa.CellField[ta.wpfloat],
+    topography_smoothed: fa.CellField[ta.wpfloat],
+    grid: icon_grid.IconGrid,
+    vertical_config: VerticalGridConfig,
+    vertical_geometry: VerticalGrid,
+    nshift: int = 0,  # shift parameter used for vertical nesting
+) -> fa.CellKField[ta.wpfloat]:
+    """
+    Computes the 3D vertical coordinate fields.
+    """
+
+    z3d_i = xp.zeros((grid.num_cells, grid.num_levels + 1), dtype=ta.wpfloat)
+
+    topography = topography.asnumpy()
+    topography_smoothed = topography_smoothed.asnumpy()
+    vct_a = vct_a.asnumpy()
+
+    dvct1 = 100.0
+    minrat1 = 1.0 / 3.0  # minimum relative layer thickness for nominal thicknesses <= dvct1 (in m)
+    dvct2 = 500.0
+    minrat2 = 0.5  # minimum relative layer thickness for a nominal thickness of dvct2
+
+    z3d_i[:, grid.num_levels] = topography
+    ktop_thicklimit = grid.num_levels * xp.ones(grid.num_cells, dtype=ta.wpfloat)
+
+    for k in range(vertical_geometry.nflatlev + 1):
+        k1 = k + nshift
+        z3d_i[:, k] = vct_a[k1]
```

Comment:
```suggestion
     z3d_i[:, range(vertical_geometry.nflatlev + 1)] = vct_a[range(vertical_geometry.nflatlev + 1+nshift)]
```
This should work as well

Reply:
`k1` should be `range(nshift, vertical_geometry.nflatlev + 1 + nshift)`
do I still edit or does this make readability worse? (same as below)

Reply:
I think it still makes sense